docs/content/guide/migration.ngdoc

```diff
@@ -13,6 +13,156 @@ which drives many of these changes.
 * Several new features, especially animations, would not be possible without a few changes.
 * Finally, some outstanding bugs were best fixed by changing an existing API.
 
+
+
+
+
+# Migrate from 1.4 to 1.5
+
+Angular 1.5 takes a big step towards enabling developers to prepare themselves for a smoother
+transition to Angular 2 in the future. Architecturing your applications using components, making use
+of lifecycle hooks in directive controllers and relying on native ES6 features (such as classes and
+arrow functions) are now all possible with Angular 1.5.
+
+
+This release includes numerous bug and security fixes as well as performance improvements to core
+services, directives, filters and helper functions. Existing applications can start enjoying the
+benefits of such changes in `$compile`, `$parse`, `$animate`, `$animateCss`, `$sanitize`, `ngOptions`,
+`currencyFilter`, `numberFilter`, `copy()` (to name but a few) without any change in code.
+
+New features have been added to more than a dozen services, directives and filters across 7 modules.
+Among them, a few stand out:
+
+* `angular.component()`: Introducing "components", a special sort of directives that are easy to
+  configure and promote best practices (plus can bring Angular 1 applications closer to Angular 2's
+  style of architecture).
+* Multi-slot transclusion: Enabling the design of more powerful and complex UI elements with a much
+  simpler configuration and reduced boilerplate.
+* `ngAnimateSwap`: A new directive in `ngAnimate`, making it super easy to create rotating
+  banner-like components
+* Testing helpers: New helper functions in `ngMock`, simplifying testing for animations, component
+  controllers and routing.
+
+Also, notable is the improved support for ES6 features, such as classes and arrow functions. These
+features are now more reliably detected and correctly handled within the core.
+
+
+All this goodness doesn't come without a price, though. Below is a list of breaking changes (grouped
+by module) that need to be taken into account while migrating from 1.4. Fortunately, the majority of
+them should have a pretty low impact on most applications.
+
+
+## Core
+
+We tried to keep the breaking changes inside the core components to a bare minimum. Still, a few of
+them were unavoidable.
+
+### Services (`$parse`)
+
+Due to [0ea53503](https://github.com/angular/angular.js/commit/0ea535035a3a1a992948490c3533bffb83235052),
+a new special property, `$locals`, will be available for accessing the locals from an expression.
+This change will only affect you, if a `$locals` property does already exist (and needs to be
```

Comment:
`This is a breaking change only, if ...` (because it's not clear that a new feature is a BC)